src/components/Shared/CopyIcon.tsx

```diff
@@ -0,0 +1,40 @@
+import { useState } from 'react'
+
+import { ContentCopy } from '@mui/icons-material';
+import { Typography } from '@mui/material';
+import IconButton from '@mui/material/IconButton';
+
+type CopyIconProps = {
+  text: string;
+  arialLabel: string;
+}
+const CopyIcon = ({
+  text,
+  arialLabel,
+}: CopyIconProps) => {
+  const [copied, setCopied] = useState<boolean>(false);
+  const handleClick = () => {
+    navigator.clipboard.writeText(text)
+    .then(() => {
+      setCopied(true);
+      setTimeout(() => setCopied(false), 1000); 
+    })
+    .catch((err) => {
+      console.error('Failed to copy text to clipboard:', err);
+    });
+  };
+  return (
+    <>
+      <IconButton size='small' aria-label={arialLabel} onClick={handleClick}>
+        <ContentCopy fontSize='inherit' />
+      </IconButton>
+      {copied && (

```

Comment:
How about placing this `copied` message above the button instead? Because on the home page, the width of the Base and Time components are smaller so `copied` spills over the next item.